yew-macro/src/html_tree/html_tag/mod.rs

```diff
@@ -136,32 +136,90 @@ impl ToTokens for HtmlTag {
         } = &attributes;
 
         let vtag = Ident::new("__yew_vtag", tag_name.span());
-        let attr_pairs = attributes.iter().map(|TagAttribute { label, value }| {
+        let attr_pairs = attributes.iter().map(|TagAttribute { label, question_mark, value }| {
             let label_str = label.to_string();
-            quote_spanned! {value.span()=> (#label_str.to_owned(), (#value).to_string()) }
-        });
-        let set_booleans = booleans.iter().map(|TagAttribute { label, value }| {
-            let label_str = label.to_string();
-            quote_spanned! {value.span()=>
-                if #value {
-                    #vtag.add_attribute(&#label_str, &#label_str);
+            if question_mark.is_some() {
+                quote_spanned! {value.span()=>
+                    {
+                        let __yew_value = ::std::option::Option::as_ref(&(#value)).map(::std::string::ToString::to_string);
+                        (::std::string::String::from(#label_str), __yew_value)
+                    }
+                }
+            } else {
+                quote_spanned! {value.span()=>
+                    {
+                        (::std::string::String::from(#label_str), ::std::option::Option::Some(::std::string::ToString::to_string(&#value)))
+                    }
                 }
             }
         });
+        let set_booleans = booleans.iter().map(
+            |TagAttribute {
+                 label,
+                 question_mark: _,
+                 value,
+             }| {
+                let label_str = label.to_string();
+                quote_spanned! {value.span()=>
+                    {
+                        if #value {
+                            #vtag.add_attribute(#label_str, &#label_str);
+                        }
+                    }
+                }
+            },
+        );
         let set_kind = kind.iter().map(|kind| {
-            quote_spanned! {kind.span()=> #vtag.set_kind(&(#kind)); }
+            let value = &kind.value;
+            if kind.question_mark.is_some() {
+                quote_spanned! {value.span()=>
+                    {
+                        if let ::std::option::Option::Some(__yew_kind) = ::std::option::Option::as_ref(&(#value)) {
```

Comment:
```suggestion
                        if let ::std::option::Option::Some(ref __yew_kind) = #value {
```

If this decreases the quality of the error messages, please don't include it.

Reply:
In this case it would make sense to use the syntax because it's more concise. I marked this as resolved because it does unfortunately generate a much less useful error message.